---a/net/minecraft/item/ItemBed.java
+++b/net/minecraft/item/ItemBed.java
@@ -11,69 +11,54 @@
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.World;
 
-public class ItemBed extends Item
-{
+public class ItemBed extends Item {
+
     private static final String __OBFID = "CL_00001771";
 
-    public ItemBed()
-    {
+    public ItemBed() {
         this.setCreativeTab(CreativeTabs.tabDecorations);
     }
 
-    public boolean onItemUse(ItemStack stack, EntityPlayer playerIn, World worldIn, BlockPos pos, EnumFacing side, float hitX, float hitY, float hitZ)
-    {
-        if (worldIn.isRemote)
-        {
+    public boolean onItemUse(ItemStack stack, EntityPlayer playerIn, World worldIn, BlockPos pos, EnumFacing side, float hitX, float hitY, float hitZ) {
+        if (worldIn.isRemote) {
             return true;
-        }
-        else if (side != EnumFacing.UP)
-        {
+        } else if (side != EnumFacing.UP) {
             return false;
-        }
-        else
-        {
+        } else {
             IBlockState iblockstate = worldIn.getBlockState(pos);
             Block block = iblockstate.getBlock();
             boolean flag = block.isReplaceable(worldIn, pos);
 
-            if (!flag)
-            {
+            if (!flag) {
                 pos = pos.up();
             }
 
-            int i = MathHelper.floor_double((double)(playerIn.rotationYaw * 4.0F / 360.0F) + 0.5D) & 3;
+            int i = MathHelper.floor_double((double) (playerIn.rotationYaw * 4.0F / 360.0F) + 0.5D) & 3;
             EnumFacing enumfacing1 = EnumFacing.getHorizontal(i);
             BlockPos blockpos1 = pos.offset(enumfacing1);
-            boolean flag1 = block.isReplaceable(worldIn, blockpos1);
+            boolean flag1 = worldIn.getBlockState(blockpos1).getBlock().isReplaceable(worldIn, blockpos1); // CraftBukkit
             boolean flag2 = worldIn.isAirBlock(pos) || flag;
             boolean flag3 = worldIn.isAirBlock(blockpos1) || flag1;
 
-            if (playerIn.canPlayerEdit(pos, side, stack) && playerIn.canPlayerEdit(blockpos1, side, stack))
-            {
-                if (flag2 && flag3 && World.doesBlockHaveSolidTopSurface(worldIn, pos.down()) && World.doesBlockHaveSolidTopSurface(worldIn, blockpos1.down()))
-                {
+            if (playerIn.canPlayerEdit(pos, side, stack) && playerIn.canPlayerEdit(blockpos1, side, stack)) {
+                if (flag2 && flag3 && World.doesBlockHaveSolidTopSurface(worldIn, pos.down()) && World.doesBlockHaveSolidTopSurface(worldIn, blockpos1.down())) {
                     int j = enumfacing1.getHorizontalIndex();
                     IBlockState iblockstate1 = Blocks.bed.getDefaultState().withProperty(BlockBed.OCCUPIED, Boolean.valueOf(false)).withProperty(BlockBed.FACING, enumfacing1).withProperty(BlockBed.PART, BlockBed.EnumPartType.FOOT);
 
-                    if (worldIn.setBlockState(pos, iblockstate1, 3))
-                    {
+                    if (worldIn.setBlockState(pos, iblockstate1, 3)) {
                         IBlockState iblockstate2 = iblockstate1.withProperty(BlockBed.PART, BlockBed.EnumPartType.HEAD);
                         worldIn.setBlockState(blockpos1, iblockstate2, 3);
                     }
 
                     --stack.stackSize;
                     return true;
-                }
-                else
-                {
+                } else {
                     return false;
                 }
-            }
-            else
-            {
+            } else {
                 return false;
             }
         }
     }
-}
+}
